src/apispec/ext/marshmallow/field_converter.py

```diff
@@ -328,23 +319,13 @@ def field2length(self, field, **kwargs):
         min_attr = "minItems" if is_array else "minLength"
         max_attr = "maxItems" if is_array else "maxLength"
 
-        for validator in validators:
-            if validator.min is not None:
-                if hasattr(attributes, min_attr):
```

Comment:
IIUC, the `hasattr` test was wrong and should have been written

```py
    if "min" in validator
```

This code was broken and the function would always return the last validator min since `hasattr` would always return `False`.

This PR fixes that bug.